desktop_version/src/FileSystemUtils.cpp

```diff
@@ -141,9 +145,37 @@ char *FILESYSTEM_getUserLevelDirectory()
 	return levelDir;
 }
 
+static unsigned char cast_to_unsigned_char(char orig) {
+    return static_cast<unsigned char>(orig);
+}
+
 void FILESYSTEM_loadFileToMemory(const char *name, unsigned char **mem,
                                  size_t *len, bool addnull)
 {
+        if (strcmp(name, "levels/special/stdin.vvvvvv") == 0) {
+            // this isn't *technically* necessary when piping directly from a file, but checking for that is annoying
+            static bool STDIN_LOADED = false;
+            static std::vector<char> STDIN_BUFFER;
```

Comment:
My C++ is a little rough, but would using `<unsigned char>` here make it so we don't need the `std::transform` at the bottom? This seems like a really unnecessary memory duplication because of C++-isms; frankly I'd prefer the longer C version if the tradeoff is that the actual code in execution is simpler. (I suppose you could also use a local `char* dst` and then do `*mem = (unsigned char*) dst` at the end, the compiler shouldn't complain there...)

Either way, this is _way_ better, it's a really neat idea!

Reply:
That second idea is smart, I'll try that.